src/app/shared/state/entry-wizard.service.ts

```diff
@@ -0,0 +1,141 @@
+
+  /**
+   * Updates the list of git registries
+   */
+  updateGitRegistries() {
+    this.entryWizardStore.setLoading(true);
+    this.defaultService
+      .getUserRegistries()
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (registries: Array<string>) => {
+          this.entryWizardStore.update({ gitRegistries: registries });
+        },
+        (error: HttpErrorResponse) => {
+          this.alertService.detailedError(error);
+        }
+      );
+  }
+
+  /**
+   * Updates the list of organizations based on the given git registry
+   * @param registry git registry
+   */
+  updateGitOrganizations(registry: string) {
+    this.entryWizardStore.setLoading(true);
+    this.entryWizardStore.update({ gitOrganizations: undefined, gitRepositories: undefined });
+    const registryEnum = this.convertSourceControlStringToEnum(registry);
+    this.defaultService
+      .getUserOrganizations(registryEnum)
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (organizations: Array<string>) => {
+          this.entryWizardStore.update({ gitOrganizations: organizations });
+        },
+        (error: HttpErrorResponse) => {
+          this.alertService.detailedError(error);
+        }
+      );
+  }
+
+  /**
+   * Updates the list of repositories based on the given git registry and organization
+   * @param registry git registry
+   * @param organization git organization
+   */
+  updateGitRepositories(registry: string, organization: string) {
+    this.entryWizardStore.setLoading(true);
+    const registryEnum = this.convertSourceControlStringToEnum(registry);
+
+    this.defaultService
+      .getUserOrganizationRepositories(registryEnum, organization)
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (repositories: Array<Repository>) => {
+          this.entryWizardStore.update({ gitRepositories: repositories });
+        },
+        (error: HttpErrorResponse) => {
+          this.alertService.detailedError(error);
+        }
+      );
+  }
+
+  /**
+   * Adds a workflow for the given git repository
+   * @param registry git registry
+   * @param organization git organization
+   * @param repositoryName git repository name
+   */
+  addWorkflow(registry: string, organization: string, repositoryName: string) {
+    this.entryWizardStore.setLoading(true);
+    const registryEnum = this.convertSourceControlStringToEnum(registry);
+    this.defaultService
+      .addWorkflow(registryEnum, organization, repositoryName)
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (workflow: BioWorkflow) => {
+          this.alertService.detailedSuccess('Workflow ' + registry + '/' + organization + '/' + repositoryName + ' has been added');
+        },
+        (error: HttpErrorResponse) => {
+          this.alertService.detailedError(error);
```

Comment:
If this fails, don't you need to update the UI? Won't the toggle show the repo enabled, but it actually won't be?

Reply:
Yes this is a good point.